File: problema3.hs
import Data.List (transpose)

-- Matriz original
-- Matriz original
matrizX :: [[Int]]
matrizX = [ [1, 2, 3]
          , [4, 5, 6]
          , [7, 8, 9]
          ]


-- La transpuesta en Haskell ya existe en Data.List
-- Pero podemos implementarla con funciones lambda:
transpuestaLambda :: [[a]] -> [[a]]
transpuestaLambda ([]:_) = []
transpuestaLambda matriz = map head matriz : transpuestaLambda (map tail matriz)

matrizTranspuestaLambda :: [[Int]]
matrizTranspuestaLambda = transpuestaLambda matrizX


-- Función auxiliar para imprimir matrices
imprimirMatriz :: Show a => [[a]] -> IO ()
imprimirMatriz = mapM_ print


main :: IO ()
main = do
    putStrLn "Matriz original X:"
    imprimirMatriz matrizX
    
    putStrLn "\nMatriz transpuesta XT (usando función lambda):"
    imprimirMatriz matrizTranspuestaLambda
    

File: problema3.js
// Matriz original X (lista de listas)
const X = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
];


// función lambda para calcular transpuesta
const transpuesta=(matriz)=> {
    //obtener numero de filas y columnas
    const filas = matriz.length
    const columnas = matriz[0].length
    //crear matriz transupuesta usando map
    return Array.from({length:columnas}, (_,i)=> matriz.map(fila => fila[i]))
}

const XT = transpuesta(X)


// Mostrar resultados
console.log('Matriz original X:');
X.forEach(fila => console.log(fila));

console.log('\nMatriz transpuesta XT:');
XT.forEach(fila => console.log(fila));